app/controllers/api/v1/group_order_articles_controller.rb

```diff
@@ -0,0 +1,95 @@
+class Api::V1::GroupOrderArticlesController < Api::V1::BaseController
+  include Concerns::CollectionScope
+
+  before_action :require_ordergroup
+  before_action :require_minimum_balance, only: [:create, :update] # destroy is ok
+  before_action :require_enough_apples, only: [:create, :update] # destroy is ok
+  # @todo allow decreasing amounts when minimum balance isn't met
+
+  def index
+    render_collection search_scope
+  end
+
+  def show
+    goa = scope.find(params.require(:id))
+    render_goa_with_oa(goa)
+  end
+
+  def create
+    GroupOrderArticle.transaction do
+      oa = order_articles_scope.find(create_params.require(:order_article_id))
+      go = current_ordergroup.group_orders.find_or_create_by!(order_id: oa.order_id)
+      goa = go.group_order_articles.create!(order_article: oa)
+      goa.update_quantities((create_params[:quantity] || 0).to_i, (create_params[:tolerance] || 0).to_i)
+      oa.update_results!
```

Comment:
shouldn't the model instead of the controller be responsible for updating `order_articles` and `group_order`?

Reply:
Yes.
I'm a little fearful touching and refactoring the member ordering update logic. Keeping it separate like this was a first step, without having to deal with the 'classic' member ordering system. Perhaps I'd rather wait until one of the two directions starts to fade, and clean it up for one use-case.

I'll have a look at it some time later, to see if I can wrap my head around it a bit better.

Reply:
Moved to #579. There are at least three places where group order articles are saved.
I'm somewhat in favour of taking that up separately, if you're ok.

Reply:
I looked it it again, and the moment when `OrderArticle#update_results!` and `GroupOrder#update_price!` are called depends on the action that is taken: if multiple products are updated in a group order, `GroupOrder#update_price!` only needs to be called once at the end (not each time a GroupOrderArticle is modified) - this happens in the regular member ordering screens. And when an OrderArticle is updated among multiple group orders, `#update_results!` only needs to be called once at the end too (we don't do this currently, as far as I know, but it could be possible).

Perhaps ideally I'd have some kind of system that queues `OrderArticle#update_results!` and `GroupOrder#update_price!` for the speciic order article and group order when a GroupOrderArticle is changed, and then at the end of the transaction, all of these queued updates are performed once. We don't have such a system currently. So I think the place for these calls is in the code that requests the GroupOrderArticle updates, which in this case is the controller.